src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -474,16 +453,22 @@ private static bool IsUsableHttp11Connection(HttpConnection connection, long now
                     }
                 }
 
-                if (IsUsableHttp11Connection(connection, nowTicks, _poolManager.Settings._pooledConnectionLifetime, async))
+                if (connection.LifetimeExpired(nowTicks, _poolManager.Settings._pooledConnectionLifetime))
                 {
-                    if (NetEventSource.Log.IsEnabled()) connection.Trace("Found usable connection in pool.");
-                    return new ValueTask<HttpConnection?>(connection);
+                    if (NetEventSource.Log.IsEnabled()) connection.Trace("Found expired connection in pool.");
+                    connection.Dispose();
+                    continue;
                 }
-                else
+
+                if (!connection.PrepareForReuse(async))
                 {
                     if (NetEventSource.Log.IsEnabled()) connection.Trace("Found invalid connection in pool.");
                     connection.Dispose();
+                    continue;
                 }
```

Comment:
The bodies of these if blocks are identical except for one word in the traced message.  If that distinction isn't important, we could combine these:
```C#
if (connection.LifetimeExpired(nowTicks, _poolManager.Settings._pooledConnectionLifetime) ||
    !connection.PrepareForReuse(async))
{
    if (NetEventSource.Log.IsEnabled()) connection.Trace("Found expired or invalid connection in pool.");
    connection.Dispose();
    continue;
}
```

Reply:
The distinction seems important to me.